tests/test_plugin_features.py

```diff
@@ -41,55 +178,40 @@ def test_create_plugin():
     assert np.allclose(circ_times_area, ds["circ"] * ds["area_um"])
 
 
-def test_remove_all_plugin_features():
-    plugin_path = get_plugin_file()
-    _ = dclab.load_plugin_feature(plugin_path)
-
-    ds = dclab.new_dataset(retrieve_data("rtdc_data_hdf5_rtfdc.zip"))
-    assert "circ_per_area" in ds
-    assert "circ_times_area" in ds
-
-    remove_all_plugin_features()
-
-    assert "circ_per_area" not in ds
-    assert "circ_times_area" not in ds
-
-
-def test_plugin_metadata():
-    plugin_path = get_plugin_file()
-    plugin_list = dclab.load_plugin_feature(plugin_path)
-    pf1, pf2 = plugin_list
-
-    plugin_file_info = find_plugin_feature_script(plugin_path)
-
-    assert pf1.plugin_metadata == plugin_file_info
-    assert pf2.plugin_metadata == plugin_file_info
+def test_pf_import_plugin_fail():
+    bad_plugin_path = "not/a/real/path/plugin.py"
+    with pytest.raises(PluginImportError):
+        _ = dclab.load_plugin_feature(bad_plugin_path)
 
 
-def compute_single_plugin_feature(rtdc_ds):
-    circ_per_area = rtdc_ds["circ"] / rtdc_ds["area_um"]
-    return circ_per_area
-
-
-def compute_multiple_plugin_features(rtdc_ds):
-    circ_per_area = rtdc_ds["circ"] / rtdc_ds["area_um"]
-    circ_times_area = rtdc_ds["circ"] * rtdc_ds["area_um"]
-    return {"circ_per_area": circ_per_area, "circ_times_area": circ_times_area}
+def test_pf_initialize_plugin_after_loading():
+    h5path = retrieve_data("rtdc_data_hdf5_rtfdc.zip")
+    with dclab.new_dataset(h5path) as ds:
+        circ_per_area = compute_single_plugin_feature(ds)
+    with h5py.File(h5path, "a") as h5:
+        h5["events"]["circ_per_area"] = circ_per_area
+    with dclab.new_dataset(h5path) as ds:
+        assert "circ_per_area" not in ds
+        plugin_path = ''
+        other_info = {}
+        feature_label = ""
+        is_scalar = True
+        feature_info = example_plugin_feature_info()
+        PlugInFeature(feature_label, is_scalar, plugin_path,
+                      other_info, **feature_info)
+        assert "circ_per_area" in ds
```

Comment:
also: `assert "circ_per_area" in ds.features_innate`

Reply:
Bit confused here. do you mean `assert "circ_per_area" not in ds.features_innate`?

Reply:
I probably meant *before* creation of the plugin.